### PR TITLE
feat(multi tenant): Copy Site with Content Types Inclusion #28035

### LINKED ISSUE #28035: Copy Site with Content Types Inclusion
### Parent
#27727

### User Story
As a dotCMS user, I want to be able to include content types when copying a site, so that I can easily replicate the structure of the original site.

### Acceptance Criteria
- Add a checkbox in the dialog of copy site options to include content types.
- When the checkbox is selected, all content types associated with the site should be copied over to the new site.
- The copied content types should retain their configurations and settings in the new site.
- By default, copying Content Types when copying Sites is disabled. In order to enable it, set the `FEATURE_FLAG_ENABLE_CONTENT_TYPE_COPY` to `true`.

### Proposed Objective
Static content: "Core Features"

### Proposed Priority
Static content: "Priority 3 - Average"

### External Links
Static Content: "N/A"

### Assumptions & Initiation Needs
- The user has permission to copy sites in dotCMS.
- The user understands the relationship between content types and sites in dotCMS.

### Quality Assurance Notes & Workarounds
- QA should verify that the checkbox for including content types works as expected.
- QA should test the copied content types to ensure they retain their configurations.
- QA must test the Relationship field and all of its combinations.

### Sub-Tasks & Estimates
- Develop checkbox in the copy site dialog (2 hours)
- Implement functionality to copy content types (4 hours)
- Test

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes

**NOTE TO DEVS: The intention of this PR is to provide the actual code changes that are required to implement this feature, and some minor improvements in logging cleanup. Further code refactoring, including lowering code complexity, single-letter variables and such, will be handled in a separate PR.** 

* Allows Users to copy Content Types when creating a Site based on an existing one. Relationships and related contents are updated as well.
* Adds missing Javadoc.
* Changes several logging statements to `DEBUG` level in order to minimize excessive logging.
* Adds more meaningful logging when copying a Site, specifying what items are being processed, and how many. Moreover, `DEBUG` logging can be activated in case more information is needed.

### CODE REVIEW

File: dotCMS/src/enterprise/java/com/dotcms/enterprise/priv/HostAssetsJobImpl.java [hunk 1]
```diff
@@ -905,8 +943,25 @@ private void processToCopyContentletDependencies(final List<Contentlet> contents
...
-                        destinationContent).new ContentletRelationshipRecords(r, true);
+				ContentletRelationshipRecords related;
+				if (sourceContent.getContentTypeId().equals(destinationContent.getContentTypeId())) {
+					related = new ContentletRelationships(
+							destinationContent).new ContentletRelationshipRecords(r, true);
+				} else {
+					final ContentTypeMapping contentTypeMapping =
+							copiedContentTypes.get(sourceContent.getContentTypeId());
+					final Field relationshipField =
+							APILocator.getContentTypeFieldAPI().byContentTypeIdAndVar(contentTypeMapping.destinationContentType.id(), r.getChildRelationName());
```
Comment (jdotcms) [inline dotCMS/src/enterprise/java/com/dotcms/enterprise/priv/HostAssetsJobImpl.java:954]: rename r

Comment (jcastro-dotcms) [reply dotCMS/src/enterprise/java/com/dotcms/enterprise/priv/HostAssetsJobImpl.java:954]: I didn't want to rename any more code in order to keep this PR as clean as possible. Those renames and further refactoring will be taken care of in another PR.

File: dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java [hunk 2]
```diff
@@ -520,17 +487,50 @@ public int countForSites(final String condition, final BaseContentType base, fin
...
+  @Override
+  @EnterpriseFeature(licenseLevel = LicenseLevel.PROFESSIONAL, errorMsg = "An enterprise license is required in order to use this feature.")
+  public ContentType copyFrom(final CopyContentTypeBean copyContentTypeBean, final Host destinationSite) throws DotDataException, DotSecurityException {
     final ContentType sourceContentType = copyContentTypeBean.getSourceContentType();
-    final ContentTypeBuilder builder = ContentTypeBuilder.builder(sourceContentType).name(copyContentTypeBean.getName())
-            .fixed(false).system(false)
-            .id(null).modDate(new Date()).variable(null);
+    final ContentTypeBuilder builder = ContentTypeBuilder.builder(sourceContentType)
+            .name(copyContentTypeBean.getName())
+            .description("Copy - " + sourceContentType.description())
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java:528]: here may be not need this

**Review by jdotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java]: Done

**Pull request merged**